### PR TITLE
DTSPB-1576: Lookup lev record by system number

### PR DESCRIPTION
**Pull request by nigeldunne**
JIRA link (if applicable)

https://tools.hmcts.net/jira/browse/DTSPB-1576
Change description

Lookup lev record by system number

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java [hunk 1]
```diff
@@ -30,4 +34,14 @@
         lifeEventService.verifyDeathRecord(caseDetails, securityUtils.getSecurityDTO());
         return ResponseEntity.ok(callbackResponseTransformer.updateTaskList(request));
     }
+
+    @PostMapping(path = "/updateWithSystemNumber")
+    public ResponseEntity<CallbackResponse> updateWithSystemNumber(@RequestBody CallbackRequest request) {
+        final Integer systemNumber = request.getCaseDetails().getData().getDeathRecordSystemNumber();
+        final DeathRecord deathRecord = lifeEventService.getDeathRecordById(systemNumber);
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java]: perhaps these calls belong together in their own service with the transformer call?

File: src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java [hunk 2]
```diff
@@ -42,6 +42,26 @@ private DeathRecord mapDeathRecord(V1Death v1Death) {
...
+
+        return builder.build();
+    }
+
+    @SuppressWarnings("squid:S2583")
+    public uk.gov.hmcts.probate.model.ccd.raw.DeathRecord mapDeathRecordCCD(V1Death v1Death) {
+        if (null == v1Death) {
+            return null;
+        }
+        final uk.gov.hmcts.probate.model.ccd.raw.DeathRecord.DeathRecordBuilder builder
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java]: can the classes here be imported to make it more readable?

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java (hunk 1 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java]: Done

File: src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java (hunk 2 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java]: Refactored as discussed

**Review by sanjayparekh-hmcts — APPROVED** [review]

**Pull request closed without merging**